python/cucim/src/cucim/skimage/segmentation/_chan_vese.py

```diff
@@ -429,6 +429,7 @@ def chan_vese(
         energies = []
     phivar = tol + 1
 
+    dt = cp.asarray(dt, dtype=float_dtype)
```

Comment:
LGTM to me also.

Just FYI, I'll note that in some places using `float(dt)` may also make sense to ensure "weak" promotion handling (i.e. float scalars behave like they used to in most cases on both NumPy 1 and 2).
I don't think this needs a fix here!

Reply:
Yeah I think here we realized that we were passing a Python `float` to [a CuPy-based custom kernel]( https://github.com/grlee77/cucim/blob/cc787fd64d3e5e97b96eddcc4c64c120dd910c50/python/cucim/src/cucim/skimage/segmentation/_chan_vese.py#L107-L109 ) and relying on whatever default promotion it was doing, which CuPy changed recently. There's more context in 3 under comment: https://github.com/rapidsai/cucim/issues/742#issuecomment-2218492642

So we decided it was better that we determine the type here instead. Hence the change

Hopefully that context is helpful

Reply:
Ah, NumPy 2 + CuPy 13.2 interaction might make promotion a bit more tricky even, so I guess the Python float thing may not work there.

Reply:
Yeah actually in the case where the error was seen, we are using the default value for `dt` (as it is unspecified)

https://github.com/rapidsai/cucim/blob/fbca7fa847f8bf9916ea013dd274d88be43c99fb/python/cucim/src/cucim/skimage/segmentation/tests/test_chan_vese.py#L46

Looking at the default value for `dt`, it appears to be `0.5`

https://github.com/rapidsai/cucim/blob/fbca7fa847f8bf9916ea013dd274d88be43c99fb/python/cucim/src/cucim/skimage/segmentation/_chan_vese.py#L282

So perhaps CuPy has decided to always promote `float` to `float64` before CUDA kernel calls

Greg did the investigation here. So likely knows more about the issue

Reply:
All, good, I don't think it's worth to track this down exactly.  Eventually (probably with CuPy 14), this shouldn't be necessary for Python floats at least.  But it is definitely the pragmatic thing right now.